crates/uv-resolver/src/version_map.rs

```diff
@@ -84,16 +97,23 @@ impl VersionMap {
             NoBinary::None => false,
             NoBinary::All => true,
             NoBinary::Packages(packages) => packages.contains(package_name),
+        }; // Check if source distributions are allowed for this package.
+        let no_build = match no_build {
+            NoBuild::None => false,
+            NoBuild::All => true,
+            NoBuild::Packages(packages) => packages.contains(package_name),
         };
         Self {
             inner: VersionMapInner::Lazy(VersionMapLazy {
                 map,
                 simple_metadata,
                 no_binary,
+                no_build,
                 index: index.clone(),
                 tags: tags.clone(),
                 python_requirement: python_requirement.clone(),
                 exclude_newer: exclude_newer.copied(),
+                allowed_yanks: allowed_yanks.clone(),
```

Comment:
Maybe we should only include the versions here? `AllowedYanks` is `FxHashMap<PackageName, FxHashSet<Version>>` but we only need the versions since we're looking at a single package name.